feature/qos/ate_tests/qos_policy_config_test/qos_policy_config_test.go

```diff
@@ -1146,3 +1164,580 @@ func testCiscoSchedulerPoliciesConfig(t *testing.T) {
+				termID              string
+				dscpSet             []uint8
+				targetGroup         string
+				queueName           string
+			}{
+				desc:                "Input Classifier Type IPV6",
+				inputClassifierType: oc.Input_Classifier_Type_IPV6,
+				classifier:          "dscp_based_classifier_ipv6",
+				classType:           oc.Qos_Classifier_Type_IPV6,
+				termID:              "0",
+				targetGroup:         "target-group-BE1",
+				dscpSet:             []uint8{0, 1, 2, 3},
+				queueName:           "0",
+			})
+	}
+	dp := dut.Port(t, "port1")
+	i := q.GetOrCreateInterface(dp.Name())
+	i.SetInterfaceId(dp.Name())
+	if *deviations.ExplicitInterfaceRefDefinition {
+		i.GetOrCreateInterfaceRef().Interface = ygot.String(dp.Name())
+		i.GetOrCreateInterfaceRef().Subinterface = ygot.Uint32(0)
+	}
+	ip := &oc.Interface{Name: ygot.String(dp.Name())}
+	ip.Type = oc.IETFInterfaces_InterfaceType_ethernetCsmacd
+	s := ip.GetOrCreateSubinterface(0)
+
+	t.Logf("qos input classifier config cases: %v", cases)
+	for _, tc := range cases {
+		t.Run(tc.desc, func(t *testing.T) {
+			if tc.classType == oc.Qos_Classifier_Type_IPV4 {
+				s4 := s.GetOrCreateIpv4()
+				s4.Enabled = ygot.Bool(true)
+			} else if tc.classType == oc.Qos_Classifier_Type_IPV6 {
+				s6 := s.GetOrCreateIpv6()
+				s6.Enabled = ygot.Bool(true)
+			}
+			gnmi.Replace(t, dut, gnmi.OC().Interface(dp.Name()).Config(), ip)
+			c := i.GetOrCreateInput().GetOrCreateClassifier(tc.inputClassifierType)
+			c.SetType(tc.inputClassifierType)
+			c.SetName(tc.classifier)
+			gnmi.Replace(t, dut, gnmi.OC().Qos().Config(), q)
+		})
+
+		// Verify the Classifier is applied on interface by checking the telemetry path state values.
+		classifier := gnmi.OC().Qos().Interface(dp.Name()).Input().Classifier(tc.inputClassifierType)
+		if got, want := gnmi.Get(t, dut, classifier.Name().State()), tc.classifier; got != want {
+			t.Errorf("classifier.Name().State(): got %v, want %v", got, want)
+		}
+		if got, want := gnmi.Get(t, dut, classifier.Type().State()), tc.inputClassifierType; got != want {
+			t.Errorf("classifier.Name().State(): got %v, want %v", got, want)
+		}
+	}
+
+}
+
+func testJuniperSchedulerPoliciesConfig(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	d := &oc.Root{}
+	q := d.GetOrCreateQos()
+	dp := dut.Port(t, "port2")
+	i := q.GetOrCreateInterface(dp.Name())
+	i.SetInterfaceId(dp.Name())
+	if *deviations.ExplicitInterfaceRefDefinition {
+		i.GetOrCreateInterfaceRef().Interface = ygot.String(dp.Name())
+		i.GetOrCreateInterfaceRef().Subinterface = ygot.Uint32(0)
+	}
+
+	schedulers := []struct {
+		desc        string
+		sequence    uint32
+		priority    oc.E_Scheduler_Priority
+		inputID     string
+		inputType   oc.E_Input_InputType
+		weight      uint64
+		queueName   string
+		targetGroup string
+	}{{
+		desc:        "scheduler-policy-BE1",
+		sequence:    uint32(1),
+		priority:    oc.Scheduler_Priority_UNSET,
+		inputID:     "BE1",
+		inputType:   oc.Input_InputType_QUEUE,
```

Comment:
the inputType is always QUEUE and so doesn't need to be parameterized in the struct. It can just be inlined

Reply:
addressed